### ISSUE TITLE
Add getIntialPropsOnce to preserve the props.

### ISSUE THREAD

**Issue by SarKurd** [zeit/next.js#10746]
# Feature request

Preserve the `getIntialProps` props during client navigation(Link or push route) so we don't need to call getInitialProps again

## Is your feature request related to a problem? Please describe.

`getIntialProps` gets called again each time you navigate to that page which is totally unnecessary in most situation (for example calling an API to fetch a list of articles), the only solution at this moment i found is to use a global state.

## Describe the solution you'd like

Another static method may be, `getInitialPropsOnce` to avoid introducing breaking changes.

## Describe alternatives you've considered

Couldn't think of any better alternative.

## Additional context

None.

**Comment by ijjk** [comment]
Hi, this sounds related to our [SSG RFC](https://github.com/zeit/next.js/issues/9524). SSG should cover this use case as `getStaticProps` allows users to prerender pages/data at build time so that it's always available and used during client navigations

**Comment by SarKurd** [comment]
`getStaicProps` gets called during build time only as you said so the props never change until you build again.

This proposal is to add a runtime method like `getServerProps` to get called only once unless you refresh.

**Comment by timneutkens** [comment]
It's unlikely that we'll introduce a new method. Most likely the underlying case that you're trying to cover will be covered by `getStaticProps`'s `revalidate` option just fine.

**Comment by SarKurd** [comment]
`revalidate` gonna be amazing but not sure how it can cover the same case. gonna keep tracking the new releases, i hope to hear about it soon.

Closing this.

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
